decidim-proposals/app/controllers/decidim/proposals/collaborative_drafts_controller.rb

```diff
@@ -0,0 +1,220 @@
+            attachment: form(AttachmentForm).from_params({})
+          )
+        end
+      end
+
+      def create
+        enforce_permission_to :create, :collaborative_draft
+        @step = :step_3
+        @form = form(CollaborativeDraftForm).from_params(params)
+
+        CreateCollaborativeDraft.call(@form, current_user) do
+          on(:ok) do |collaborative_draft|
+            flash[:notice] = I18n.t("proposals.collaborative_drafts.create.success", scope: "decidim")
+
+            redirect_to Decidim::ResourceLocatorPresenter.new(collaborative_draft).path
+          end
+
+          on(:invalid) do
+            flash.now[:alert] = I18n.t("proposals.collaborative_drafts.create.error", scope: "decidim")
+            render :complete
+          end
+        end
+      end
+
+      def edit
+        enforce_permission_to :edit, :collaborative_draft, collaborative_draft: @collaborative_draft
+
+        @form = form(CollaborativeDraftForm).from_model(@collaborative_draft)
+      end
+
+      def update
+        enforce_permission_to :edit, :collaborative_draft, collaborative_draft: @collaborative_draft
+
+        @form = form(CollaborativeDraftForm).from_params(params)
+        UpdateCollaborativeDraft.call(@form, current_user, @collaborative_draft) do
+          on(:ok) do |collaborative_draft|
+            flash[:notice] = I18n.t("proposals.collaborative_drafts.update.success", scope: "decidim")
+            redirect_to Decidim::ResourceLocatorPresenter.new(collaborative_draft).path
+          end
+
+          on(:invalid) do
+            flash.now[:alert] = I18n.t("proposals.collaborative_drafts.update.error", scope: "decidim")
+            render :edit
+          end
+        end
+      end
+
+      def request_access
+        RequestAccessToCollaborativeDraft.call(@collaborative_draft, current_user) do
+          on(:ok) do |_collaborative_draft|
+            flash[:notice] = t("access_requested.success", scope: "decidim.proposals.collaborative_drafts.requests")
+          end
+
+          on(:invalid) do
+            flash.now[:alert] = t("access_requested.error", scope: "decidim.proposals.collaborative_drafts.requests")
+          end
+        end
+        redirect_to Decidim::ResourceLocatorPresenter.new(@collaborative_draft).path
+      end
+
+      def request_accept
+        AcceptAccessToCollaborativeDraft.call(@collaborative_draft, current_user, requester_user) do
+          on(:ok) do
+            flash[:notice] = t("accepted_request.success", scope: "decidim.proposals.collaborative_drafts.requests", user: requester_user.nickname)
+          end
+
+          on(:invalid) do
+            flash.now[:alert] = t("accepted_request.error", scope: "decidim.proposals.collaborative_drafts.requests", user: requester_user.nickname)
+          end
+        end
+        redirect_to Decidim::ResourceLocatorPresenter.new(@collaborative_draft).path
+      end
+
+      def request_reject
```

Comment:
All the `request_*` action should be moved to a different controller.

Reply:
ok